api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -110,6 +122,12 @@ type AzureManagedControlPlaneSpec struct {
 	// +optional
 	NetworkPlugin *string `json:"networkPlugin,omitempty"`
 
+	// NetworkPluginMode is the mode the network plugin should use.
+	// Allowed value is "overlay".
```

Comment:
I assume this means that there's only one possible value? In that case, what happens with the `// +optional` and `omitempty` tags?

Reply:
These allowed values are only enforced when the value is non-nil, so this is the behavior:
```
networkPluginMode: "overlay" # ok

networkPluginMode: null # ok

networkPluginMode: "" # error
```

Reply:
Gotcha, are there any other cases where we do validation like this instead of with the webhook? I'm fine to proceed as is, but figured it's worth noting that it might be confusing to have validation in more than one place.

Reply:
Not sure what you mean? The kubebuilder annotation here is the only thing defining this specific validation.

Reply:
Yeah, I think the general principle is

1. Prefer yaml schema validation (this is limited, but if your validation can be express in terms of kubebuilder pragmas then this is better)
2. Only do one or the other (both is worse)
3. When validation is trickier, use webhooks